biodatasets/pubmed_qa/pubmed_qa.py

```diff
@@ -0,0 +1,245 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import json
+import glob
+import datasets
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, Iterator, Tuple
+
+import utils.parsing as parsing
+import utils.schemas as schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_CITATION = """\
+@inproceedings{jin2019pubmedqa,
+  title={PubMedQA: A Dataset for Biomedical Research Question Answering},
+  author={Jin, Qiao and Dhingra, Bhuwan and Liu, Zhengping and Cohen, William and Lu, Xinghua},
+  booktitle={Proceedings of the 2019 Conference on Empirical Methods in Natural Language Processing and the 9th International Joint Conference on Natural Language Processing (EMNLP-IJCNLP)},
+  pages={2567--2577},
+  year={2019}
+}
+"""
+
+_DATASETNAME = "pubmed_qa"
+
+_DESCRIPTION = """\
+PubMedQA is a novel biomedical question answering (QA) dataset collected from PubMed abstracts.
+The task of PubMedQA is to answer research biomedical questions with yes/no/maybe using the corresponding abstracts. 
+PubMedQA has 1k expert-annotated, 61.2k unlabeled and 211.3k artificially generated QA instances. 
+
+Each PubMedQA instance is composed of:
+  (1) a question which is either an existing research article title or derived from one, 
+  (2) a context which is the corresponding abstract without its conclusion, 
+  (3) a long answer, which is the conclusion of the abstract and, presumably, answers the research question, and
+  (4) a yes/no/maybe answer which summarizes the conclusion.
+
+PubMedQA is the first QA dataset where reasoning over biomedical research texts, 
+especially their quantitative contents, is required to answer the questions.
+"""
+
+_HOMEPAGE = "https://github.com/pubmedqa/pubmedqa"
+_LICENSE = "MIT License"
+
+_URLS = {
+    "pqaa": "https://drive.google.com/uc?export=download&id=1kaU0ECRbVkrfjBAKtVsPCRF6qXSouoq9",
+    "pqal": "https://drive.google.com/uc?export=download&id=1kQnjowPHOcxETvYko7DRG9wE7217BQrD",
+    "pqau": "https://drive.google.com/uc?export=download&id=1q4T_nhhj8UvJ9JbZedhkTZHN6ZeEZ2H9",
+}
+
+_SUPPORTED_TASKS = [Tasks.QUESTION_ANSWERING]
+_SOURCE_VERSION = "1.0.0"
+_BIGBIO_VERSION = "1.0.0"
+
+_CLASS_NAMES = [
+    "Yes",
+    "No",
+    "Maybe"
+]
+
+class PubmedQADataset(datasets.GeneratorBasedBuilder):
+    """PubmedQA Dataset"""
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        # PQAU & PQAA Source
+        BigBioConfig(
+            name="pqaa_source",
```

Comment:
can we make the names and subset_ids consistent with the other dataset loader naming conventions? 
e.g. `pubmed_qa_a_source`

Reply:
Sure, I will update it into `pubmed_qa_artificial_*`, `pubmed_qa_labelled_*`, and `pubmed_qa_unlabelled_* `